### DISCUSSION TITLE
Add Entity Relationship Selectors to Entity

### DISCUSSION THREAD

**Discussion by MikeRyanDev** [ngrx/platform#3080] [Ideas]
Internally we have developed common patterns for creating selectors that help resolve entity relationships. We should consider making them generic and folding them into Entity.

**Comment by bryanrideshark** [comment]
Has anyone considered using something similar to JSONGraph? We use it internally for some things and find that it's helpful when having relationships between entities.

**Comment by dsebastien** [comment]
Can you share more details about this?
It the idea to denormalize data when taking it out of the store using selectors?

**Comment by Nosfistis** [comment]
Are there any active plans for this? It's a cool major feature, and causing a lot of custom wiring which contradicts the ease of use @ngrx/data provides.

**Comment by satanTime** [comment]
Hi guys, I was struggling with the same issue and implemented a lib that tries to solve it: 
https://www.npmjs.com/package/ngrx-entity-relationship

let me know your thoughts.

**Reply by qdelettre** [reply]
Hello there

This looks nice. What's the status of the support of the latests ngrx version for this library @satanTime ? Do you have a roadmap or need help ?

Is there a more suited solution provided by ngrx since your last message ?

Thanks

**Reply by satanTime** [reply]
Hi there. It is. I need to update it, but postponed it, because was looking for the best way to release updates for all major ngrx versions. However, still not sure whether I should create a branch for each major, or somehow to create a mono repo with all of them. 

Let me know if you have an idea / experience about that. I'll be more than happy to discuss them.

**Reply by qdelettre** [reply]
Unfortunatly i don't. Sorry. Thanks for the quick reply.
